internal/drivers/google/driver.go

```diff
@@ -500,3 +537,34 @@ func getInstanceName(runner, pool string) string {
 	name := strings.ToLower(fmt.Sprintf("%s-%s-%s-%s", namePrefix, pool, uniuri.NewLen(8), randStr)) //nolint:gomnd
 	return substrSuffix(name, maxInstanceNameLen)
 }
+
+func shouldRetry(err error) bool {
+	switch e := err.(type) {
+	case *googleapi.Error:
+		// Retry on 429 and 5xx, according to
+		// https://cloud.google.com/storage/docs/exponential-backoff.
+		return e.Code == 429 || (e.Code >= 500 && e.Code < 600)
+	case interface{ Temporary() bool }:
+		return e.Temporary()
+	default:
+		return false
+	}
+}
+
+func retry[T any](ctx context.Context, attempts int, sleep int, f func() (T, error)) (result T, err error) {
```

Comment:
Nit: specify unit for sleep time like "sleepSecs". Just makes it easier to use the function

Reply:
done